public/app/features/dimensions/editors/ResourcePicker.tsx

```diff
@@ -1,224 +1,93 @@
-                    onChangeSearch(v);
-                    setSearchQuery(v);
-                  }}
-                />
-              </Field>
-            </>
-          )}
-          {source?.value === 'url' && (
-            <Field label="URL">
-              <Input onChange={(e) => setNewValue(e.currentTarget.value)} value={newValue} />
-            </Field>
-          )}
-        </div>
-        <div className={styles.iconContainer}>
-          <Field label="Preview">
-            <div className={styles.iconPreview}>
-              {mediaType === 'icon' && <SVG src={imgSrc} className={styles.img} />}
-              {mediaType === 'image' && newValue && <img src={imgSrc} className={styles.img} />}
+    <PopoverController content={popoverElement} hideAfter={300}>
+      {(showPopper, hidePopper, popperProps) => {
+        return (
+          <>
+            {pickerTriggerRef.current && (
+              <Popover
+                {...popperProps}
+                referenceElement={pickerTriggerRef.current}
+                onMouseLeave={hidePopper}
+                onMouseEnter={showPopper}
+                onKeyDown={(event: any) => closePopover(event, hidePopper)}
+              />
+            )}
+
+            <div ref={pickerTriggerRef} onClick={showPopper} onMouseLeave={hidePopper}>
+              <InlineFieldRow className={styles.pointer}>
+                <InlineField label={null} grow>
+                  <Input
+                    value={props.niceName}
+                    placeholder={props.placeholder}
+                    readOnly={true}
+                    prefix={props.srcPath && <SVG src={props.srcPath} className={styles.icon} />}
+                    suffix={<Button icon="times" variant="secondary" fill="text" size="sm" onClick={props.onClear} />}
+                  />
+                </InlineField>
+              </InlineFieldRow>
             </div>
-          </Field>
-          <Label>{shortName}</Label>
-        </div>
-      </div>
-      {source?.value === 'folder' && filteredIndex && (
-        <div className={styles.cardsWrapper}>
-          <ResourceCards cards={filteredIndex} onChange={(v) => setNewValue(v)} value={newValue} />
-        </div>
-      )}
-
-      <Modal.ButtonRow>
-        <Button variant="secondary" onClick={() => setOpen(false)}>
-          Cancel
-        </Button>
-        <Button variant={newValue && newValue !== value ? 'primary' : 'secondary'} onClick={() => onChange(newValue)}>
-          Select
-        </Button>
-      </Modal.ButtonRow>
-      {/* TODO: add file upload
-          {tabs[1].active && (
-          <FileUpload
-            onFileUpload={({ currentTarget }) => console.log('file', currentTarget?.files && currentTarget.files[0])}
-            className={styles.tabContent}
-          />
-        )} */}
-    </div>
+          </>
+        );
+      }}
+    </PopoverController>
   );
 };
 
 const getStyles = stylesFactory((theme: GrafanaTheme2) => {
```

Comment:
Please refactor stylesFactory to useStyles2. stylesFactory is should be used only when styles getter depend on some properties other than theme.